From: Mart Raudsepp <leio@gentoo.org>
Date: Thu, 13 Dec 2018 20:29:05 +0200
Subject: notification: bind mount /etc/ld.so.cache to the sandbox

This is especially important for libstdc++ on distributions that
don't have it directly in a libdir and the runtime linker doesn't
look where needed without /etc/ld.so.cache (e.g. if libstdc++ is
in a GCC per-version subdirectory handled via /etc/ld.so.conf.d/).

If /etc/ld.so.cache is not available, the runtime linker will look
only at a set of predetermined paths - as seen with LD_DEBUG=libs
added to the bwrap call with "--setenv LD_DEBUG libs":

find library=libstdc++.so.6 [0]; searching
 search cache=/etc/ld.so.cache
 search path=/lib64:/usr/lib64          (system search path)
  trying file=/lib64/libstdc++.so.6
  trying file=/usr/lib64/libstdc++.so.6

followed by:

/usr/bin/totem-video-thumbnailer: error while loading shared libraries: libstdc++.so.6: cannot open shared object file: No such file or directory

If /etc/ld.so.cache is available, it will use that for the paths:

find library=libstdc++.so.6 [0]; searching
 search cache=/etc/ld.so.cache
  trying file=/usr/lib/gcc/x86_64-pc-linux-gnu/8.2.0/libstdc++.so.6

By bind mounting just that file out of /etc, we get it to work on
such a system.

Signed-off-by: Simon McVittie <smcv@debian.org>
Forwarded: https://github.com/flatpak/xdg-desktop-portal/pull/289
---
 src/notification.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/notification.c b/src/notification.c
index 8b9f38a..3126201 100644
--- a/src/notification.c
+++ b/src/notification.c
@@ -432,6 +432,7 @@ add_bwrap (GPtrArray *array,
   add_args (array,
             BWRAP,
             "--ro-bind", "/usr", "/usr",
+            "--ro-bind", "/etc/ld.so.cache", "/etc/ld.so.cache",
             NULL);
 
   /* These directories might be symlinks into /usr/... */
